app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -239,12 +283,26 @@ private void failGame() {
         isPlaying = false;
         LevelManager.setCurrentLevelScore(score);
         NavigationHelper.navigateToFragment(activity, new GameFailureFragment());
+        LevelConfig currentLevel = LevelManager.getGameLevels().get(LevelManager.getCurrentLevelIndex());
+        if (currentLevel.getLevelType() == LevelType.INFINITE) {
+            updateScoreInMaxScoreLeaderboard();
+        }
+        updateObstacleLeaderboards();
+    }
+
+    private void updateScoreInMaxScoreLeaderboard() {
+        String leaderboardId = getResources().getString(R.string.leaderboard_best_score);
+        long maxLevelScore = LevelManager.getLevelMaxScore(LevelManager.getCurrentLevelIndex());
+        long maxBestScore = Math.max(ScoreManager.getScore(leaderboardId), maxLevelScore);
```

Comment:
I am confused here: I thought the max score is only for infinite levels?

Reply:
This method is called only inside infinite levels.